src/OpenTelemetry.Exporter.Geneva/TLDExporter/TldExporter.cs

```diff
@@ -124,11 +129,11 @@ protected static void Serialize(EventBuilder eb, string key, object value)
                 string repr;
                 try
                 {
-                    repr = Convert.ToString(value, CultureInfo.InvariantCulture);
+                    repr = Convert.ToString(value, CultureInfo.InvariantCulture) ?? string.Empty;
                 }
                 catch
                 {
-                    repr = $"ERROR: type {value.GetType().FullName} is not supported";
+                    repr = $"ERROR: type {value!.GetType().FullName} is not supported";
```

Comment:
I think this can throw an exception if `value = null!`

Reply:
We could but we would have to pay for a `null` check for every field serialized. I went and checked, seems like none of the upstream code actually will ever pass a `null` (also added the assert to validate) so this at the end of the day is just being defensive. Even with a `null` check users could do something silly like this...

```csharp
activity.SetTag("MyTag", new MyType());

class MyType
{
   public override string? ToString() => null;
}
```

...which would cause the `Convert.ToString` call there to return `null` even though `value` itself was something real! Spooky 👻

Reply:
okay, if you checked the upstream code then I'm fine continuing as is. :)